doc/source/ray-core/examples/gentle_walkthrough.ipynb

```diff
@@ -0,0 +1,777 @@
+     ]
+    }
+   ],
+   "source": [
+    "start = time.time()\n",
+    "object_references = [\n",
+    "    retrieve_task.remote(item) for item in range(8)\n",
+    "]\n",
+    "data = ray.get(object_references)\n",
+    "print_runtime(data, start)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "8df4087c",
+   "metadata": {},
+   "source": [
+    "Running the task in parallel requires two minor code modifications.\n",
+    "To execute your Ray task remotely, you must use a `.remote()` call.\n",
+    "Ray executes remote tasks asynchronously, even on a local cluster.\n",
+    "The items in the `object_references` list in the code snippet do not directly contain the results.\n",
+    "If you check the Python type of the first item using `type(object_references[0])`,\n",
+    "you see that it is actually an `ObjectRef`.\n",
+    "These object references correspond to _futures_ for which you need to request the result.\n",
+    "The call `ray.get(...)` is for requesting the result. Whenever you call remote on a Ray task,\n",
+    "it immediately returns one or more object references.\n",
+    "Consider Ray tasks as the primary way of creating objects.\n",
+    "The following section is an example that links multiple tasks together and allows\n",
+    "Ray to pass and resolve the objects between them."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2373ddd9",
+   "metadata": {},
+   "source": [
+    "Let's review the previous steps.\n",
+    "You started with a Python function, then decorated it with `@ray.remote`, making the function a Ray task.\n",
+    "Instead of directly calling the original function in the code, you called `.remote(...)` on the Ray task.\n",
+    "Finally, you retrieved the results from the Ray cluster using `.get(...)`.\n",
+    "Consider creating a Ray task from one of your own functions as an additional exercise."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": []
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "e008a500",
+   "metadata": {},
+   "source": [
+    "Let's review the performance gain from using Ray tasks.\n",
+    "On most laptops the runtime is around 0.71 seconds,\n",
+    "which is slightly more than the slowest subtask, which is 0.7 seconds.\n",
+    "You can further improve the program by leveraging more of Ray’s API."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "54f53644",
+   "metadata": {},
+   "source": [
+    "### Object Stores\n",
+    "\n",
+    "The retrieve definition directly accesses items from the `database`. While this works well on a local Ray cluster, consider how it functions on an actual cluster with multiple computers.\n",
+    "A Ray cluster has a head node with a driver process and multiple worker nodes with worker processes executing tasks.\n",
```

Comment:
```suggestion
    "In a Ray cluster, one node has a driver process while all nodes can have worker processes that execute tasks.\n",
```

Reply:
Is this a satisfactory rewrite, @jjyao?

Reply:
hmm, yeah, this is strictly speaking correct, but keep in mind that this is an intro fro beginners.

Reply:
ok, done